docs/src/main/tut/methods.md

```diff
@@ -0,0 +1,51 @@
+---
+menu: methods
+weight: 115
+title: HTTP Methods
+---
+
+For a REST API, your service will want to support different verbs/methods.
+Http4s has a list of all the [methods] you're familiar with, and a few more.
+
+```tut:book
+import fs2.Task
+import io.circe.generic._
+import io.circe.syntax._
+import org.http4s._, org.http4s.dsl._
+import org.http4s.circe._
+
+@JsonCodec case class TweetWithId(id: Int, message: String)
+@JsonCodec case class Tweet(message: String)
+
+def getTweet(tweetId: Int): Task[Option[TweetWithId]] = ???
+def addTweet(tweet: Tweet): Task[TweetWithId] = ???
+def updateTweet(id: Int, tweet: Tweet): Task[Option[TweetWithId]] = ???
+def deleteTweet(id: Int): Task[Unit] = ???
+
+implicit val tweetWithIdEncoder = jsonEncoderOf[TweetWithId]
+implicit val tweetDecoder = jsonOf[Tweet]
+
+val tweetService = HttpService {
+  case GET -> Root / "tweets" / IntVar(tweetId) =>
+    getTweet(tweetId)
+      .flatMap(_.fold(NotFound())(Ok(_)))
+  case req @ POST -> Root / "tweets" =>
+    req.as[Tweet].flatMap(addTweet).flatMap(Ok(_))
+  case req @ PUT -> Root / "tweets" / IntVar(tweetId) =>
+    req.as[Tweet]
+      .flatMap(updateTweet(tweetId, _))
+      .flatMap(_.fold(NotFound())(Ok(_)))
+  case req @ HEAD -> Root / "tweets" / IntVar(tweetId) =>
```

Comment:
I would omit `req @` from the branches where it's not used. We want to encourage precise usage rather than boilerplate.